silverstripe/silverstripe-blog
# Fix: Publish BlogProfileImage even if Name unchanged

## Description

Due to the order in `onBeforeWrite`, the `BlogProfileImage` added in `BlogMemberExtension` doesn't get automatically published unless you have also changed the member's name (unlikely).

This is because if `FirstName` or `Surname` are unchanged, then `onBeforeWrite` returns, before being able to publish the profile image.

This PR moves the image publish lines to the top of the method so they always get run.

## Manual testing steps

1. Create a new user in Security.
2. Click **Create**
3. **Upload** a new "Blog profile image" for the user.
4. Click **Save**

**What I expect to happen:** The image gets published.
**What actually happens:** The image stays as "Draft".
## Issues

- #748

## Pull request checklist

    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
